### DISCUSSION TITLE
Is the my-test-package intended to be delivered along with the setuptools package?

### DISCUSSION THREAD

**Discussion by dus7eh** [pypa/setuptools#4488] [General]
In recent release 71.0.1 one of previous changes to exclude package data and tests was reverted: https://github.com/pypa/setuptools/commit/1a52f11e3c28b3776a5d5184c536a280f7061acd

Is this intended? I'm asking since SCA tools starting with this version detect such test package (`my-test-package`) as a dependency with unknown License.  
This raises some dependency management issues and I wanted to know if this is how you intend to proceed.

**Comment by dus7eh** [comment]
Any maintainers who could answer to my question?

**Comment by SamuelHans** [comment]
Would like to know this too with `my-test-package`.

**Comment by abravalheri** [comment]
Could you please clarify this better? Setuptools itself has no dependency on `my-test-package`. Worst case scenario it is a test fixture with no further meaning.

Probably a false positive in the tool that is generating the report.

**Comment by dus7eh** [comment]
For the sake of clarity check out screenshots below showing filesystem changes when installing two different versions of setuptools. (it is based on a python:3.11 base image)

`setuptools==71.0.0`
![Screenshot from 2024-07-31 23-28-03](https://github.com/user-attachments/assets/d1cf72a1-607c-4060-86a4-082cb4d8c3a8)

`setuptools==71.0.1`
![Screenshot from 2024-07-31 23-25-35](https://github.com/user-attachments/assets/833f6a5c-be2d-47d3-8be1-4235a746599d)

Yes, it seems to be a test package which has a regular package structure and .egg file. That's why it's treated by SCA tools as a python package. It is flagged since it has an unknown licensing set in metadata along with other dummy data.

If this is how you're going to deliever I can somewhat live with that. However be aware that it will most likely show up in SBOMs and SCA scans as an independent component.

**Reply by webknjaz** [reply]
These should definitely be included in sdists, but probably not wheels.

cc @abravalheri ^

**Reply by abravalheri** [reply]
@webknjaz, including them in the wheel, seems to be a deliberate decision from https://github.com/pypa/setuptools/pull/4479 (even if temporarily due to an existing limitstion).

**Reply by webknjaz** [reply]
Oh, I didn't realize. Thanks for pointing out!
